### PR TITLE
See highlight in Snapshot

### PR DESCRIPTION
**Pull request by hyu**
- make sure the highlight is passed into the Snapshot state
- clear the highlight when navigating between resources

### CODE REVIEW

File: web/src/LogPane.tsx [hunk 1]
```diff
@@ -53,6 +53,52 @@ class LogPane extends Component<LogPaneProps, LogPaneState> {
...
+    if (this.props.highlightsEnabled && this.props.modalIsOpen) {
+      document.removeEventListener(
+        "selectionchange",
+        this.handleSelectionChange
+      )
+    }
+
+    if (
+      this.props.highlightsEnabled &&
+      prevProps.modalIsOpen &&
```
Comment (jazzdan) [inline web/src/LogPane.tsx]: Does this logic work for the initial view? When there is no prevProps?

File: web/src/LogPane.tsx [hunk 2]
```diff
@@ -53,6 +53,52 @@ class LogPane extends Component<LogPaneProps, LogPaneState> {
     }
   }
 
+  componentDidUpdate(prevProps: LogPaneProps) {
+    if (this.props.highlightsEnabled && this.props.modalIsOpen) {
+      document.removeEventListener(
```
Comment (nicks) [inline web/src/LogPane.tsx]: in general, it's going to be pretty racey/error-prone to add/remove listeners in componentDidUpdate.

have you considered adding these checks to handleSelectionChange, and simply doing nothing in that case?

File: web/src/LogPane.tsx [hunk 3]
```diff
@@ -53,6 +53,52 @@ class LogPane extends Component<LogPaneProps, LogPaneState> {
...
+      !this.props.modalIsOpen
+    ) {
+      document.addEventListener("selectionchange", this.handleSelectionChange, {
+        passive: true,
+      })
+    }
+
+    if (this.props.name !== prevProps.name) {
+      let selection = document.getSelection()
+      this.props.handleClearHighlight()
```
Comment (nicks) [inline web/src/LogPane.tsx]: the way this bounces control back and forth between the child component and the parent component seems hard to reason about and hard to test.

have you considered having the HUD do this in the history.listen handler? that seems a lot closer to what you want - clearing the selection every time the URL changes.

Comment (hyu) [reply web/src/LogPane.tsx]: Oh, I was looking for something which would reliably fire when navigating between resources. But didn't realize it was right here. Thanks for pointing it out!

File: web/src/LogPane.tsx (hunk 2 above)
Comment (hyu) [reply web/src/LogPane.tsx]: Good point! This simplifies the code nicely.

File: web/src/LogPane.tsx (hunk 1 above)
Comment (hyu) [reply web/src/LogPane.tsx]: I changed this around per Nick's advice below, so this piece is no longer relevant.

**Pull request merged**